src/_pytest/logging.py

```diff
@@ -425,6 +424,22 @@ def _log_cli_enabled(self):
             "--log-cli-level"
         ) is not None or self._config.getini("log_cli")
 
+    @pytest.hookimpl(hookwrapper=True, tryfirst=True)
+    def pytest_collection(self):
+        # This has to be called before the first log message is logged,
+        # so we can access the terminal reporter plugin.
+        self._setup_cli_logging()
+
+        # TODO write to file support needed?
```

Comment:
@thisch any reason not to? If there isn't I good reason, I would prefer this to go out feature-complete rather than need another iteration.

Reply:
Because adding support for log-to-file support is trickier. Simply copying the ctx manager that are used in pytest_runtestloop does not work.

Reply:
I have a working implementation now. It was necessary to remove the `closing` context manager.